cmd/agent/app/options/options.go

```diff
@@ -198,6 +207,17 @@ func (o *GrpcProxyAgentOptions) Validate() error {
 	if err := validateAgentIdentifiers(o.AgentIdentifiers); err != nil {
 		return fmt.Errorf("agent address is invalid: %v", err)
 	}
+	if o.ServerLeaseSelector != "" {
+		if _, err := labels.Parse(o.ServerLeaseSelector); err != nil {
+			return fmt.Errorf("invalid server count lease selector: %w", err)
+		}
```

Comment:
We should probably have a check that if o.ServerLeaseSelector != "" then o.KubeconfigPath != "".

Reply:
Done.

Reply:
I disagree. If you pass an empty kubeconfig to `clientcmd.BuildConfigFromFlags` then it defaults to the in-cluster config, which is probably what most users want.

Reply:
Good point

Reply:
Reverted!